crates/ruff_text_size/src/traits.rs

```diff
@@ -76,3 +76,24 @@ where
         T::range(self)
     }
 }
+
+/// A slice of the source text.
+pub trait TextSlice: Sealed {
+    /// Returns the slice of the text within the given `range`.
+    ///
+    /// ## Note
+    ///
+    /// This is the same as `&self[range]` if `self` is a `str` and `range` a `TextRange`.
+    ///
+    /// ## Panics
+    /// If the range is out of bounds.
+    fn slice(&self, range: impl Ranged) -> &str;
+}
+
+impl Sealed for str {}
+
+impl TextSlice for str {
+    fn slice(&self, ranged: impl Ranged) -> &str {
+        &self[ranged.range()]
+    }
+}
```

Comment:
hmm, most of the uses I see of this method in this PR seem to be cases where we could just directly slice the string using `[]` syntax. Not sure I see the use of this trait.

Reply:
It's correct, there aren't too many usages of `slice` in this PR but `slice` is a very common operator on `Locator` and this trait could help us migrate these calls more easily. But I'm also fine with removing it for now. 

Ideally we would implement `std::ops::Index` for `T` where `T: Ranged` but we can't do that for a foreign trait :( 

An other alternative is to implement `Ranged::slice(str)` although I think that looks off on the call site

```
expression.slice(source)
```

Reply:
I also don't mind the trait too much! Just wondered if it was necessary. If it makes the transition easier, that's good enough reason for me